### ISSUE TITLE
Bad inference (3)

### ISSUE THREAD

**Issue by keyssson** [alexiglad/EBT#14]
Dear author, this time I tried fineweb-10BT + baseline_model_xxs and trained for 1000000 steps. Yet the inference on gsm8k was still terrible. The  accuracy was 0 and generation was nothing. So I wonder what exactly went wrong. Was the training step not enough(cause I saw the val_loss was not quite converge)? Or was the dataset too small(I saw the default was fineweb-100BT)?  

<img width="1095" height="800" alt="Image" src="https://github.com/user-attachments/assets/7e0ec2c3-4095-412b-a588-4ac939c3c25b" />

<img width="906" height="266" alt="Image" src="https://github.com/user-attachments/assets/438ac8fc-ac6a-4deb-b8a6-037b59a957a3" />

<img width="904" height="263" alt="Image" src="https://github.com/user-attachments/assets/d3a419d7-3965-400f-925e-d7ff528a33be" />

<img width="865" height="1019" alt="Image" src="https://github.com/user-attachments/assets/4543f774-5d7b-4af2-a01d-0237d8aff68b" />

**Comment by alexiglad** [comment]
Can you share which code you've changed? I see a lot of diffs, so I'm wondering if something was changed that could have impacted the inference. Seems like something is going on causing the generation to be completely random.

**Comment by keyssson** [comment]
I only modified a little bit of the code. And I don't think I have changed the important part. Maybe I just upload some files? Which part you think was the most likely part that went wrong? Pretrain or inference? 
[base_model_trainer.py](https://github.com/user-attachments/files/22574121/base_model_trainer.py)
[train_model.py](https://github.com/user-attachments/files/22574122/train_model.py)
[fineweb_dataloader.py](https://github.com/user-attachments/files/22574123/fineweb_dataloader.py)

**Comment by alexiglad** [comment]
@keyssson can you push your code so we can see diffs compared to the main branch?

**Comment by alexiglad** [comment]
Closing this due to no response.
